core/configuration/dt_config_provider.go

```diff
@@ -171,6 +173,15 @@ func validateConfiguration(config *DtConfiguration) error {
 }
 
 func generateAgentId() int64 {
-	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	var rng *rand.Rand
+
+	var seed [8]byte
+	_, err := cryptoRand.Read(seed[:])
+	if err != nil {
+		rng = rand.New(rand.NewSource(time.Now().UnixNano()))
+	} else {
+		rng = rand.New(rand.NewSource(int64(binary.LittleEndian.Uint64(seed[:]))))
```

Comment:
Why the Uint -> int conversion? Could we interpret as int in the first place?

Reply:
This would be functionally equivalent:

```go
var seedInt int64
binary.Read(bytes.NewBuffer(seed[:]), binary.LittleEndian, &seedInt)
```

Though internally that implementation does literally the same thing: `int64(binary.LittleEndian.Uint64(buf))`. So doesn't seem like an improvement to me.